trinity/utils/ipc.py

```diff
@@ -0,0 +1,188 @@
+import functools
+from multiprocessing.connection import (
+    Listener,
+    Client,
+    wait,
+)
+import operator
+import os
+import signal
+import threading
+import time
+import uuid
+
+
+def wait_for_ipc(ipc_path, timeout=1):
+    start_at = time.time()
+    while time.time() - start_at < timeout:
+        if os.path.exists(ipc_path):
+            break
```

Comment:
Wow that's a tight loop. Maybe a tiny `else: sleep(0.05)` is reasonable to avoid slamming disk/cpu.